compiler/rustc_ast_passes/src/ast_validation.rs

```diff
@@ -292,9 +297,18 @@ impl<'a> AstValidator<'a> {
     }
 
     fn check_trait_fn_not_const(&self, constness: Const) {
-        if let Const::Yes(span) = constness {
-            self.dcx().emit_err(errors::TraitFnConst { span });
-        }
+        let Const::Yes(span) = constness else {
+            return;
+        };
+
+        // FIXME(const_trait_impl): If the trait or impl is not const and feature `const_trait_impl`
+        // is enabled, provide a structured suggestion to make the trait (impl) const.
```

Comment:
I don't feel like performing ugly `Span` manipulations today to find the proper spot in between `impl` and `Trait` (while respecting defaultness, polarity and unsafety), otherwise I would've included this in this PR...